tests/ui/rfcs/rfc-3573-is/name-resolution-errors.rs

```diff
@@ -0,0 +1,27 @@
+//! Test that the lexical scope for `is`'s bindings doesn't extend farther than it should.
+//@ edition: 2024
+//@ aux-crate: is_macro=is-macro.rs
+#![feature(builtin_syntax)]
+use is_macro::is;
+
+fn main() {
+    // `is` used where `let` expressions can be used behaves like a `let` expression.
+    if x //~ ERROR cannot find value `x` in this scope
+        && is!(true is x)
+        && x
+    {
+        x;
+    }
+    x; //~ ERROR cannot find value `x` in this scope
+
+    // Elsewhere, `is`'s bindings only extend to the end of its `&&`-chain.
+    if x //~ ERROR cannot find value `x` in this scope
+        && (is!(true is x) && x)
+        && x //~ ERROR cannot find value `x` in this scope
+        && (is!(true is x) || x) //~ ERROR cannot find value `x` in this scope
```

Comment:
```suggestion
        && (is!(true is x) || x) //~ ERROR cannot find value `x` in this scope
        && (!is!(true is x) || x) //~ ERROR cannot find value `x` in this scope
```

Making sure this *doesn't* work, even though it technically could, because it'd be confusing and wasn't proposed by the RFC.